packages/rest/src/rest.server.ts

```diff
@@ -310,6 +310,11 @@ export class RestServer extends Context implements Server, HttpServerLike {
         this._setupOperation(verb, path, routeSpec);
       }
     }
+
+    this.staticAssets.forEach(assetEntry => {
+      const {path, rootDir, options} = assetEntry;
+      this._httpHandler.registerStaticAssets(path, rootDir, options);
```

Comment:
Why don't we create an Express Router as the container so that `registerStaticAssets` can just add a new entry to the router without rebuilding the handler?

Reply:
I am responding which some assumptions, please clarify if I misunderstood something.

We are ultimately using an Express Router - https://github.com/strongloop/loopback-next/blob/master/packages/rest/src/router/static-assets-route.ts#L35.

We have to register the static assets router with the Routing Table so that we have the ability to check the matching route for a request in the Routing Table, and return the static assets router, if none was matched.

If I misunderstood your statement, can you clarify what you mean by "container" and "rebuilding the handler".

Reply:
I'm saying we always create an instance of `StaticAssetsRoute` as the place holder and use it to register static assets. This way we don't have to do the following:
```
this.staticAssets.push(/*...*/);
delete this._httpHandler; // I assume it's used to remove the cached _httpHandler to pick up new static assets
```

Reply:
I am still not getting it 😬.

One of the requirements of `app.static()` is that it should be able to add new routes even after the app has started. Will it work with what you are suggesting?